### ISSUE TITLE
Design System Updates: Request Type Icons

### ISSUE THREAD

**Issue by annaseulgi** [hackforla/311-data#1629]
### Overview
We need to update 311 request types to be icons due to accessibility concerns with colors. As we transition the map to have a lighter gray monochromatic background, including icons for each request type could be worth exploring to also aid with improving accessibility.

### Action Items
- [x] Discuss with design team what their thoughts are of this
- [x] Reference 311-SF website & Google for icon ideas
- [x] Assign designers to ticket
- [x] Designers can assign request types to icons (material design system will be a good resource)
- [x] Feedback/Iterations
- [x] Approval 
- [ ] Other items <>

**Comment by ryanfchase** [comment]
@pogwon @annaseulgi is this still being worked on? We will move this to In Progress if so

**Comment by pogwon** [comment]
> @pogwon @annaseulgi is this still being worked on? We will move this to In Progress if so

Yes, I'm currently working on it.

**Comment by pogwon** [comment]
Revised and updated icons have been added to Updated Design System page.

<img width="548" alt="Screenshot 2024-06-03 at 1 35 42 PM" src="https://github.com/hackforla/311-data/assets/129242704/609f2597-d936-4911-8a5f-c1f27a6c7347">

**Comment by ryanfchase** [comment]
@pogwon if this is ready for review, please do the following:
- [ ] Move ticket status to `In Review`
- [ ] Add `ready for design lead`
- [ ] Unassign

Let's also check all the boxes in the original post so its easier to track progress, thanks!

**Comment by annaseulgi** [comment]
Reviewed and moving this to Done(Without Merge) since this is not a priority item at the moment (but should be reviewed in the future).

Future tasks for this: design mockup using icon set rather than colors to indicate request types

**Comment by annaseulgi** [comment]
@pogwon hey I know you just finished a pretty big ticket for the -/+ functionality. i also know you put together this icon set :). let me know if you want to take this ticket in August (or whenever we have capacity)

**Comment by pogwon** [comment]
I was unaware of the high-fidelity wireframe deliverables, so I re-assigned myself to this issue and will be working on it. Will focus on how they will look close-up as well as from a zoomed out perspective when there are several icons clustered together. 

We also discussed potentially including icons in the service request modals so they match with the map, so I will be brainstorming potential solutions as well.

**Comment by Joy-Truex** [comment]
Hey @pogwon ! Can you move to "in progress" with a brief update when you get a chance if youre still able to work on this? Thanks!

**Comment by pogwon** [comment]
An issue I ran into while including the icons in the service request modals (to match with the map) is that they are simply too big. Making the icons smaller makes it hard to distinguish them from each other so I'm working on some new designs that are still similar enough to the current icon set.

**Comment by pogwon** [comment]
These are potentially how icons would look on the map. While I don't see an issue with the icons while zoomed in, trying to look at them while zoomed out might present an issue. It might be hard to make sense of the map because of the low contrast between the icons and the actual map. Furthermore, the white type request icons within the solid-color background of the icon circles could make it potentially straining to see against the light background of the map.  If we are transitioning to a lighter gray, monochromatic background, should we assume the blue that is used for borders will be taken out? That might be a better way to check the icons, and if they don't fit, we might have to explore alternatives for the icon designs that work within a monochromatic palette. I'll conduct some research to see if there are any other potential solutions.

<img width="1440" alt="Icons HF Zoomed In" src="https://github.com/user-attachments/assets/ee9ae93c-4a57-4fb5-806f-d85149028a15">
<img width="1440" alt="Icons HF Zoomed Out" src="https://github.com/user-attachments/assets/753a06c7-33fe-46fe-abd9-e3eef1c20e3b">

**Comment by pogwon** [comment]
After discussing with the design team, we concluded that because the whole point was to design more accessible icons, it might be worth taking a step back and reevaluate if the current designs (with the request type icons) are the right choice. They can be difficult to discern, especially when users are zoomed out, so it might be worth exploring other design options. For zoomed out views, we talked about exploring clustering icons, but I will explore other potential options.

Some research regarding clustering:

- [Google Maps: How to cluster map markers](https://mapsplatform.google.com/resources/blog/how-cluster-map-markers/)
- [MapsZen](https://mapsplatform.google.com/resources/blog/how-cluster-map-markers/) shows clustering with different colored icons
- [Discussion showing clustering with icons to show multiple statuses](https://ux.stackexchange.com/questions/76402/how-to-cluster-map-markers-with-different-statuses)

**Comment by ryanfchase** [comment]
> After discussing with the design team, we concluded that because the whole point was to design more accessible icons, it might be worth taking a step back and reevaluate if the current designs (with the request type icons) are the right choice. They can be difficult to discern, especially when users are zoomed out, so it might be worth exploring other design options. For zoomed out views, we talked about exploring clustering icons, but I will explore other potential options.
> 
> Some research regarding clustering:
> 
> * [Google Maps: How to cluster map markers](https://mapsplatform.google.com/resources/blog/how-cluster-map-markers/)
> * [MapsZen showing clustering with different colored icons](https://mapsplatform.google.com/resources/blog/how-cluster-map-markers/)
> * [Discussion showing clustering with icons to show multiple statuses](https://ux.stackexchange.com/questions/76402/how-to-cluster-map-markers-with-different-statuses)

I've been working on map clusters on another project, and this was another possible design for faceted clusters that I liked. The example doesn't have any interaction beyond basic clustering (it would be nice if you could hover over the bars on the donut charts). But still good inspiration:
- https://docs.mapbox.com/mapbox-gl-js/example/cluster-html/

**Comment by pogwon** [comment]
> > After discussing with the design team, we concluded that because the whole point was to design more accessible icons, it might be worth taking a step back and reevaluate if the current designs (with the request type icons) are the right choice. They can be difficult to discern, especially when users are zoomed out, so it might be worth exploring other design options. For zoomed out views, we talked about exploring clustering icons, but I will explore other potential options.
> > Some research regarding clustering:
> > 
> > * [Google Maps: How to cluster map markers](https://mapsplatform.google.com/resources/blog/how-cluster-map-markers/)
> > * [MapsZen showing clustering with different colored icons](https://mapsplatform.google.com/resources/blog/how-cluster-map-markers/)
> > * [Discussion showing clustering with icons to show multiple statuses](https://ux.stackexchange.com/questions/76402/how-to-cluster-map-markers-with-different-statuses)
> 
> I've been working on map clusters on another project, and this was another possible design for faceted clusters that I liked. The example doesn't have any interaction beyond basic clustering (it would be nice if you could hover over the bars on the donut charts). But still good inspiration:
> 
>     * https://docs.mapbox.com/mapbox-gl-js/example/cluster-html/

Thanks for this - I have thought about including donut charts within the clustering but am not sure if this will work if we're going to be sticking to a monochromatic design. But along with some alternatives, I will still try to design a version that uses colors to see how we feel about it and how we can optimize accessibility.

**Comment by Joy-Truex** [comment]
Hi Peter, 
I wanted to share a new map I found when I was checking to see if my family in Miami may be impacted by power outages from the hurricane. [FPL Power Tracker](https://www.fplmaps.com/)

I found it interesting to see how they are clustering data points that changes as you zoom in and out. Also take a look at the zoom controls, specifically how they disable at certain points.
Hope this helps with inspiration :)

**Comment by pogwon** [comment]
@ryanfchase An option we are exploring are larger icons (when zoomed out) that show how many 311 requests there were in the general area that will then break off into specific request icons as the user zooms in. Is this something that the dev. team can implement (are we somehow able to count the number of requests in a certain area of the map that can be then reflected in the icons)?

@Joy-Truex 

<img width="370" alt="Image" src="https://github.com/user-attachments/assets/78259cd1-9754-4185-a0b4-dfcbd5a3f439" />

**Comment by ryanfchase** [comment]
@pogwon FYI we are going to use the Questions column + `ready for dev lead` label when you have questions for us.

I'll need to confer with dev lead @traycn. ETA 2 days

**Comment by ryanfchase** [comment]
I am adding this to our Product Planning agenda, since it ties into a question I am asking all leads for input on loading data on launch. We will leave this in Questions, but I will add `ready for product`.

**Comment by ryanfchase** [comment]
Based on discussion in the Product Planning meeting, it seems this ticket is impacted by two opinions:
- design team is looking to collect evidence from users on what type of data aggregation is desirable. Each aggregation type, of which there are many, may have different design implications, so design is hesitant to guess as to which will be most useful
- design also wants to spend more time workshopping user flows for Blank Map first, this ticket is unblocked per discussion #1848 
- engineering is working on a new model for how data is ingested into the application, we're told we should wait to see how the new model is implemented before we can commit to a design

I will add these bullet points the action items shortly. Leaving this `ready for product` until then (ETA 1hr or so)

**Comment by ryanfchase** [comment]
Moving to icebox
- Designer has confirmed that the question was resolved
- dependencies are outlined here
- added `Needs More Info` because we should update the following before this gets worked on again
  - Overview (needs X for Y structure)
  - Iterations (needs to migrate comments with iteration images into original ticket)
